app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/EditPortalLocales.vue

```diff
@@ -0,0 +1,152 @@
+export default {
+  components: {
+    LocaleItemTable,
+    AddLocale,
+  },
+  mixins: [alertMixin],
+  data() {
+    return {
+      isAddLocaleModalOpen: false,
+      lastPortalSlug: undefined,
+      alertMessage: '',
+    };
+  },
+  computed: {
+    ...mapGetters({
+      uiFlags: 'portals/uiFlagsIn',
+    }),
+    currentPortalSlug() {
+      return this.$route.params.portalSlug;
+    },
+    currentPortal() {
+      return this.$store.getters['portals/portalBySlug'](
+        this.currentPortalSlug
+      );
+    },
+    locales() {
+      return this.currentPortal.config.allowed_locales;
+    },
+    allowedLocales() {
+      return Object.keys(this.locales).map(key => {
+        return this.locales[key].code;
+      });
+    },
+  },
+  mounted() {
+    this.lastPortalSlug = this.currentPortalSlug;
+  },
+  methods: {
+    changeDefaultLocale({ localeCode }) {
+      this.updatePortalLocales({
+        allowedLocales: this.allowedLocales,
+        defaultLocale: localeCode,
+        successMessage: this.$t(
+          'HELP_CENTER.PORTAL.CHANGE_DEFAULT_LOCALE.API.SUCCESS_MESSAGE'
+        ),
+        errorMessage: this.$t(
+          'HELP_CENTER.PORTAL.CHANGE_DEFAULT_LOCALE.API.ERROR_MESSAGE'
+        ),
+      });
+    },
+    deletePortalLocale({ localeCode }) {
+      const updatedLocales = this.allowedLocales.filter(
+        code => code !== localeCode
+      );
+      const defaultLocale = this.currentPortal.meta.default_locale;
+      this.updatePortalLocales({
+        allowedLocales: updatedLocales,
+        defaultLocale,
+        successMessage: this.$t(
+          'HELP_CENTER.PORTAL.DELETE_LOCALE.API.SUCCESS_MESSAGE'
+        ),
+        errorMessage: this.$t(
+          'HELP_CENTER.PORTAL.DELETE_LOCALE.API.ERROR_MESSAGE'
+        ),
+      });
+    },
+    async updatePortalLocales({
+      allowedLocales,
+      defaultLocale,
+      successMessage,
+      errorMessage,
+    }) {
+      try {
+        await this.$store.dispatch('portals/update', {
+          portalSlug: this.currentPortal.slug,
+          config: {
+            default_locale: defaultLocale,
+            allowed_locales: allowedLocales,
+          },
+        });
+        this.alertMessage = successMessage;
+      } catch (error) {
+        this.alertMessage = error?.message || errorMessage;
+      } finally {
+        this.showAlert(this.alertMessage);
+      }
+    },
+    closeAddLocaleModal() {
+      this.isAddLocaleModalOpen = false;
+      this.selectedPortal = {};
+    },
+    addLocale() {
+      this.isAddLocaleModalOpen = true;
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.portal-locales {
+  padding: var(--space-small) var(--space-normal);
+  width: 100%;
+  background: var(--white);
+  height: 100%;
+  padding: 0 var(--space-medium);
+  .button-container {
+    display: flex;
+    justify-content: flex-end;
+  }
+  .locale-container {
+    margin-top: var(--space-large);
```

Comment:
@muhsin-k Can we reduce margin top, Feels like some empty space at top.

```suggestion
    margin-top: var(--space-smaller);
```